### DISCUSSION TITLE
Difficulty configuring CI_REGISTRY setting

### DISCUSSION THREAD

**Discussion by dandn** [renovatebot/renovate#17777] [Request Help] [answered]
I'm running renovate using [whitesource/renovate-on-prem](https://github.com/whitesource/renovate-on-prem/releases) v2.5.1.

I'm having an issue configuring CI_REGISTRY following the documentation [here](https://docs.renovatebot.com/modules/manager/gitlabci/). 

For a given GitLab repo I attempted to add a `registryAliases` entry by updating the `renovate.json` file as follows:
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "dependencyDashboard": true,
  "registryAliases": {
    "$CI_REGISTRY": "docker-registry.mycompany.net"
  }
}
```

This approach does not work as I get the following `Action Required: Fix Renovate Configuration` generated under GitLab issues:
```
Location: renovate.json
Error type: The renovate configuration file contains some invalid settings
Message: Invalid 'registryAliases.registryAliases.$CI_REGISTRY' configuration: value is not a url
```

Presumably I am not adding this registryAliases entry correctly.
Could anyone advise how to correct the entry within `renovate.json` ?  
(Or does this setting rather have to go in `config.js`)?

**Comment by rarkins** [comment, marked as answer]
It's expecting the URL to be in format `https://.....`. That validation might be too strict.

You can try in `config.js` because Renovate doesn't perform strict validation there, but of course it looks like something needs fixing.

**Reply by dandn** [reply]
Yes indeed. In my example above, using `https://docker-registry.mycompany.net` will not generate the renovate error.
However, prepending the `https` results in incorrectly formatted docker images.

I will attempt to move this to the `config.js` and report back.
In the meantime, should I escalate this to an issue? (At the very least something to be clarified in the documentation?)

**Reply by rarkins** [reply]
Yes, please create an issue. At the minimum we have a docs problem, but perhaps we need to loosen the registryAliases validation
